src/components/task/uc/deleteUserData.uc.unit.test.js

```diff
@@ -0,0 +1,332 @@
+
+describe('in "deleteUserData.uc" the function', () => {
+	describe('deleteUserData', () => {
+		const userId = new ObjectId();
+		const replaceUserId = new ObjectId();
+
+		// true result
+		it('should return a function that return an array.', async () => {
+			const result = deleteUserRelatedData();
+			expect(result).to.be.an('array').with.lengthOf(5);
+			expect(result[0], 'where first element is a function').to.be.a('function');
+			expect(result[1], 'where second element is a function').to.be.a('function');
+			expect(result[2], 'where third element is a function').to.be.a('function');
+			expect(result[3], 'where fourth element is a function').to.be.a('function');
+			expect(result[4], 'where fourth element is a function').to.be.a('function');
+		});
+
+		describe('deletePrivateSubmissions', () => {
+			const initStubs = ({ studentId }) => {
+				const deleteSingleSubmissionsFromUserStub = sinon.stub(repo, 'deleteSingleSubmissionsFromUser');
+				const privateSubmissions = [
+					{ _id: new ObjectId(), studentId },
+					{ _id: new ObjectId(), studentId },
+				];
+				deleteSingleSubmissionsFromUserStub.withArgs(userId).returns(getExpectedDeleteMany(2));
+
+				const findSingleSubmissionIdsByUserStub = sinon.stub(repo, 'findUserSubmissionsByUser');
+				findSingleSubmissionIdsByUserStub.callsFake(() => []);
+				findSingleSubmissionIdsByUserStub.withArgs(userId).returns(privateSubmissions);
+
+				return { findSingleSubmissionIdsByUserStub, deleteSingleSubmissionsFromUserStub, privateSubmissions };
+			};
+
+			// true work
+			it('should return deleted private submissions', async () => {
+				// given
+				const {
+					findSingleSubmissionIdsByUserStub,
+					deleteSingleSubmissionsFromUserStub,
+					privateSubmissions,
+				} = initStubs({
+					studentId: userId,
+				});
+				// when
+				const { deletePrivateSubmissions } = mapByName(deleteUserRelatedData());
+				const result = await deletePrivateSubmissions(userId);
+				findSingleSubmissionIdsByUserStub.restore();
+				deleteSingleSubmissionsFromUserStub.restore();
+				// then
+				expect(result.complete).to.be.true;
+				expect(result.trashBinData).to.be.an('object');
+				expect(result.trashBinData).to.haveOwnProperty('scope');
+				expect(result.trashBinData).to.haveOwnProperty('data');
+				expect(result.trashBinData.scope).to.be.equal('submissions-private');
+				expect(result.trashBinData.data).to.have.members(privateSubmissions);
+			});
+
+			// user not exist
+			it('should return empty data if there is no shared submission', async () => {
+				// given
+				const otherUserId = new ObjectId();
+				const { findSingleSubmissionIdsByUserStub, deleteSingleSubmissionsFromUserStub } = initStubs({
+					studentId: userId,
+				});
+				// when
+				const { deletePrivateSubmissions } = mapByName(deleteUserRelatedData());
+				const result = await deletePrivateSubmissions(otherUserId);
+				findSingleSubmissionIdsByUserStub.restore();
```

Comment:
Instead of calling `restore` on every stub in every test, we can use the [sandbox functionalty of sinon](https://sinonjs.org/releases/latest/sandbox/) and just use a afterEach which cleans all subs created: `afterEach(sinon.restore);`